lib/bin/supllib/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+if(CONFIG_FLOAT)
+  if(CONFIG_FP_HARDABI)
+    set(float_dir hard-float)
+  else()
+    assert(0 "Unreachable code")
+  endif()
+else()
+  assert(0 "Soft float not supported")
+endif()
+
+assert(GCC_M_CPU "GCC_M_CPU must be set to find correct lib")
+
+set(lib_path lib/${GCC_M_CPU}/${float_dir})
+
+set(SUPL_CLIENT_LIB_PATH ${ZEPHYR_BASE}/../supl/${lib_path})
```

Comment:
Instead of forcing it to a path, should we use an environment variable instead? CC @SebastianBoe and @bjorn-tore-taraldsen

Reply:
If we have to choose a path,  then I would probably suggest instead what @jhn-nordic proposes: `nrf/ext`

Reply:
I think forcing a path is the simples solution for the user to use, and for us to explain in the documentation. But it also give user the fewest possibilities to configure their repo by them self. So the question here is probably what are the chances that the user want this .a file to be located somewhere else.

Reply:
Ok, I thought  putting it into nrf/ext would confuse west, but that is of course for versioned files. Yes, I can move it in to that folder. So I just need conclution on foced vs environment variable.

Reply:
Wouldn't placing it inside NCS folder structure, make it easier for the user to, by accident, publish the library as part of their git repo, and by that make it public available? By doing this the user are breaking the license agreement. 
My thinking is having it located "on the side of" the repo structure could prevent this from happening.

Reply:
libsupl.a should be in the variable as well, and it should be overridable through an env var.

Reply:
Yes @bjorn-tore-taraldsen but see my comment [here](https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1792#issuecomment-579274043). Adding it to the `.gitignore` would make it relatively hard for the users to do this by mistake.

Reply:
I have now moved the location of the library to ext/supl and added a .gitignore file to ignore the supl folder. I have also removed the supl_session.h file from the nrf integration code in this PR and it will only be distributed in the .zip file that the user download.

Because I moved the expected location of the supl library into the nrf/ext/ folder, I changed the CMakeLists.txt file in that folder to look for the supl folder if the CONFIG_SUPL_CLIENT_LIB is set. And then I added a CMakeLists.txt file to the .zip file that we are going to distribute. This simplifies CMakeLists.txt files and makes them more maintainable.